### PR TITLE
Add a mungegithub tool for bulk LGTM of PRs

### PR DESCRIPTION
**Pull request by brendandburns**
This adds a tool which lists all small-ish (user configurable) PRs along with their diffs in a condensed format so that a reviewer can quickly skim through a large number of PRs and LGTM them if appropriate.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/contrib/1720)

### CODE REVIEW

**Comment by eparis** [comment]
```
$ hack/verify-all.sh -v
Skipping hack/../hack/verify-all.sh
Boilerplate header is wrong for: /home/travis/gopath/src/k8s.io/contrib/mungegithub/mungers/bulk-lgtm.go
!!! 'gofmt -s' needs to be run on the following files: 
./mungegithub/mungers/bulk-lgtm.go
!!! 'golint' problems: 
/home/travis/gopath/src/k8s.io/contrib/mungegithub/mungers/bulk-lgtm.go:41:1: comment on exported type BulkLGTM should be of the form "BulkLGTM ..." (with optional leading article)
/home/travis/gopath/src/k8s.io/contrib/mungegithub/mungers/bulk-lgtm.go:52:1: exported method BulkLGTM.Munge should have comment or be unexported
/home/travis/gopath/src/k8s.io/contrib/mungegithub/mungers/bulk-lgtm.go:90:1: exported method BulkLGTM.AddFlags should have comment or be unexported
/home/travis/gopath/src/k8s.io/contrib/mungegithub/mungers/bulk-lgtm.go:96:1: exported method BulkLGTM.Name should have comment or be unexported
/home/travis/gopath/src/k8s.io/contrib/mungegithub/mungers/bulk-lgtm.go:100:1: exported method BulkLGTM.RequiredFeatures should have comment or be unexported
/home/travis/gopath/src/k8s.io/contrib/mungegithub/mungers/bulk-lgtm.go:104:1: exported method BulkLGTM.Initialize should have comment or be unexported
/home/travis/gopath/src/k8s.io/contrib/mungegithub/mungers/bulk-lgtm.go:121:1: exported method BulkLGTM.EachLoop should have comment or be unexported
/home/travis/gopath/src/k8s.io/contrib/mungegithub/mungers/bulk-lgtm.go:125:1: exported method BulkLGTM.FindPR should have comment or be unexported
/home/travis/gopath/src/k8s.io/contrib/mungegithub/mungers/bulk-lgtm.go:137:1: exported method BulkLGTM.ServeLGTM should have comment or be unexported
/home/travis/gopath/src/k8s.io/contrib/mungegithub/mungers/bulk-lgtm.go:162:1: exported method BulkLGTM.ServePRDiff should have comment or be unexported
/home/travis/gopath/src/k8s.io/contrib/mungegithub/mungers/bulk-lgtm.go:194:1: exported method BulkLGTM.ServePRs should have comment or be unexported
Found flags in golang files not in the list of known flags. Please add these to hack/verify-flags/known-flags.txt
bulk-lgtm-changed-files
bulk-lgtm-max-commits
bulk-lgtm-max-diff
```

File: mungegithub/bulk-lgtm/www/index.html [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  <script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.5.5/angular.min.js" type="text/javascript"></script>
+  <script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.5.5/angular-animate.min.js" type="text/javascript"></script>
+  <script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.5.5/angular-aria.min.js" type="text/javascript"></script>
+  <!--
+  Local copy of angular-material.js with the fix in https://github.com/angular/material/issues/5681 applied.
+  This makes rendering the PR tab ~2X faster.
+  -->
+  <script src="angular-material.js" type="text/javascript"></script>
+  <script src="https://cdnjs.cloudflare.com/ajax/libs/moment.js/2.12.0/moment.min.js" type="text/javascript"></script>
+  <script src="https://cdnjs.cloudflare.com/ajax/libs/angular-moment/1.0.0-beta.5/angular-moment.min.js" type="text/javascript"></script>
```
Comment (eparis) [inline mungegithub/bulk-lgtm/www/index.html]: I don't see any usage of `am-time-ago` so we can, I think, drop all of the moment stuff.

File: mungegithub/bulk-lgtm/www/script.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
+"use strict";
+var app = angular.module('BulkLGTMModule', ['ngMaterial', 'angularMoment']);
```
Comment (eparis) [inline mungegithub/bulk-lgtm/www/script.js]: and this include of angularMoment

File: mungegithub/bulk-lgtm/www/script.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  $compileProvider.debugInfoEnabled(false);
+}]);
+
+app.controller('BulkLGTMCntl', ['DataService', '$interval', '$location', BulkLGTMCntl]);
+
+function BulkLGTMCntl(dataService, $interval, $location) {
+  var self = this;
+  self.prs = {};
+  self.prsCount = 0;
+  self.health = {};
```
Comment (eparis) [inline mungegithub/bulk-lgtm/www/script.js]: junk

File: mungegithub/bulk-lgtm/www/script.js [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+  self.metadata = {};
+  self.testResults = {};
+  self.tabLoaded = {};
+  self.functionLoading = {};
+  self.queryNum = 0;
+  self.selected = 1;
+  self.OverallHealth = "";
```
Comment (eparis) [inline mungegithub/bulk-lgtm/www/script.js]: junk

File: mungegithub/bulk-lgtm/www/script.js [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+  self.StatOrder = "-Count";
```
Comment (eparis) [inline mungegithub/bulk-lgtm/www/script.js]: health, testReules, tabLoaded, functionLoading, queryNum, selected, OverallHealth, StatOrder all seem to be copy/paste fluff.

File: mungegithub/bulk-lgtm/www/script.js [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+  // Request Avatars that are only as large necessary (CSS limits them to 40px)
+  function fixPRAvatars(prs) {
+    angular.forEach(prs, function(pr) {
+      if (/^https:\/\/avatars.githubusercontent.com\/u\/\d+\?v=3$/.test(pr.AvatarURL)) {
+        pr.AvatarURL += '&size=40';
+      }
+    });
+  }
+
+  function refreshPRs() {
```
Comment (eparis) [inline mungegithub/bulk-lgtm/www/script.js]: Is this used at all? Doesn't seem so. Which means fixPRAvatars is also a vestigial appendage?

File: mungegithub/bulk-lgtm/www/index.html [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    <!-- Display the white border around the indented portion -->
+    <md-content class="md-whiteframe-z2">
+      <md-toolbar layout="row" layout-align="space-around center">
+        <h1 class="md-display-2">Bulk LGTM Tool</h1>
+        <a ng-href="https://k8s.io"><img src="https://raw.githubusercontent.com/kubernetes/kubernetes/master/logo/logo.png" alt="kubernetes logo" class="titleLogo"></a>
+      </md-toolbar>
+      <md-content class="md-padding">
+        <foo ng-repeat="pr in cntl.prs">
+          <md-list-item class="md-3-line">
+            <a class="md-avatar" ng-href="{{cntl.metadata.RepoPullUrl}}{{pr.user.login}}">
```
Comment (eparis) [inline mungegithub/bulk-lgtm/www/index.html]: Does this `<a>` tag work?

File: mungegithub/bulk-lgtm/www/index.html [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+              <img ng-src="{{pr.user.avatar_url}}" alt="{{pr.user.login}}">
+            </a>
+            <md-content class="md-list-item-text" layout="column">
+              <h3 class="md-body-1">
+                <a ng-href="{{pr.html_url}}">#{{pr.number}}: {{pr.title}}</a>
+              </h3>
+              <h4 class="md-body-2">{{pr.Reason}}</h4>
+                <p>{{pr.time | date:'medium'}} (<span style="color: green">+{{pr.additions}}</span>/<span style="color: red">-{{pr.deletions}}</span>)</p>
```
Comment (eparis) [inline mungegithub/bulk-lgtm/www/index.html]: indentation is off...

**Comment by test-foxish** [comment]
recomputing cla status...

**Comment by brendandburns** [comment]
closed in favor of https://github.com/kubernetes/test-infra/pull/1752

**Pull request closed without merging**